libutils/file_lib.c

```diff
@@ -529,10 +529,11 @@ void PathWalk(
 
     for (size_t i = 0; i < n_children; i++)
     {
-        /* The basename(3) function might potentially mutate the child, but we
-         * don't mind. */
+        // The basename(3) function might potentially mutate the child.
         char *const child = SeqAt(children, i);
-        const char *const b_name = basename(child);
+        char buf[strlen(child) + 1];
```

Comment:
Sorry, looking at this again, should we really be using a variable length array here? With unlimited max size due to `strlen()`?

Isn't there some fixed constant size we can use here, like `PATH_MAX`?